### PR TITLE
Fix the implementation of Support.{remove_,}extension and update tests

### PR DESCRIPTION
**Pull request by dinosaure**
A simple fix for #16 where I decided to took the implementation from OCaml (and add the copyright). The `support` module depends on `Sys` now and the patch seems to modify the order of modules into the `module_collision` tests.

### CODE REVIEW

File: tests/complex/module_collision/reference [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [[1m[31mError[0m]: Local module conflict,
- Module [1mA[0m is provided simultaneously by [1mmodule_collision/Z/a.mli[0m and
-[1mmodule_collision/B/A.mli[0m
+ Module [1mA[0m is provided simultaneously by [1mmodule_collision/B/A.mli[0m and
+[1mmodule_collision/Z/a.mli[0m
```
Comment (Octachron) [inline tests/complex/module_collision/reference:3]: Did you run this test before your changes? It looks like a possibly unrelated change.

Comment (dinosaure) [reply tests/complex/module_collision/reference:3]: Ah yes, you are right, the error appeared before, I will split the commit in two.

**Comment by dinosaure** [comment]
I think this PR is ready. I added a documentation to precise that the semantic of `extension` is **different** than `Filename.extension` from the standard library. The current implementation of `codept` expects the extension without the dot when `Filename.extension` keeps the dot.

**Comment by Octachron** [comment]
The testsuite should be fixed now and the first commit can be dropped.

**Comment by dinosaure** [comment]
> The testsuite should be fixed now and the first commit can be dropped.

Done :+1:.

File: lib/support.mli [hunk 2]
```diff
@@ -1,7 +1,25 @@
 (** Support module for recently introduced function in stdlib *)
 
 val extension: string -> string
+(** [extension name] is the shortest suffix [ext] of [name0] where:
+    - [name0] is the longest suffix of [name] that does not contain
+      a directory separator;
+    - [ext] does {b not} starts with a period (such as
+      [name0 ^ "." ^ ext = name])
```
Comment (Octachron) [inline lib/support.mli:8]: The two conditions `shortest suffix` and `does not start with a period` do not mix well: at this point the empty word still satisfies the two conditions. I would reword it as:
```
- [.] [ext] is a suffix of [name0]
```
